### PR TITLE
[ONNX] Fix type casting for reduce ops

### PR DESCRIPTION
**Pull request by neginraoof**
Fix type casting for reduce ops in ONNX exporter. PyTorch promotes dtypes bool and all integer types to long for these ops.

This fix only covers traced modules where dtype is present

### CODE REVIEW

**Comment by KsenijaS** [comment]
looks good, but you should probably rebase your branch against master since there are some merge failures

File: torch/onnx/symbolic_opset9.py [hunk 1]
```diff
@@ -261,8 +261,19 @@ def _slice(g, input, axes, starts, ends):
     return g.op("Slice", input, axes_i=axes, starts_i=starts, ends_i=ends)
 
 
+def _maybe_cast_reduce_op_input(g, self):
+    dtype = self.type().scalarType()
+    # This check only covers traced modules where dtype is present
+    if dtype is not None:
+        # pytorch reduce-ops cast all other input types to int64
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:268]: should it be *all other integral types? float types are not included right?

**Review by BowenBao — APPROVED** [review]
Minor comment on comments, LGTM.

File: torch/onnx/symbolic_opset9.py (hunk 1 above)
Comment (neginraoof) [reply torch/onnx/symbolic_opset9.py]: Thanks for catching this. Correct, I fixed it.

**Comment by neginraoof** [comment]
@houseroad Can we merge this one? Thanks a lot!

**Review by houseroad — COMMENTED** [review]
Something is wrong with the importing script... I am retrying...

**Review by houseroad — APPROVED** [review]

**Pull request closed without merging**